app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,91 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+  before_action :convert_hyphen_to_dash, only: [:create, :update]
+
+  def index
+    render json: tool_groups_ordered_by_name, include: params[:include], fields: field_params, status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def create_tool
+    tool_group_id = params[:tool_group_id].to_i
+    resource_id = params.dig(:data, :relationships, :tool, :data, :id).to_i
+    suggestions_weight = params.dig(:data, :attributes, "suggestions-weight").to_f
+
+    resource_tool_group = ResourceToolGroup.create!(
+      tool_group_id: tool_group_id,
+      resource_id: resource_id,
+      suggestions_weight: suggestions_weight
+    )
+
+    response.headers["Location"] = "tool-groups/#{resource_tool_group.id}"
+    render json: resource_tool_group, status: :created
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def update_tool
+    existing = resource_tool_group
+    existing.update!(
+      resource_id: params[:data][:attributes]["resource-id"],
+      suggestions_weight: params[:data][:attributes]["suggestions-weight"]
+    )
+    render json: existing, status: :accepted
+  end
+
+  def delete_tool
+    if resource_tool_group&.destroy!
```

Comment:
when looking up the `resource_tool_group`, this is ignoring the tool-group id. This means the following request succeeds when it should have returned a 404 instead.

```
curl --location --request DELETE 'https://mobile-content-api-stage.cru.org/tool-groups/invalid/tools/2' \
--header 'Accept: application/vnd.api+json' \
```

Reply:
this should be solved now